### ISSUE TITLE
MMBN6 Wireless & online connection fails

### ISSUE THREAD

**Issue by ssbmars** [visualboyadvance-m/visualboyadvance-m#457]
We play Megaman Battle Network 6 online pvp in VBA-Link 1.8 using Hamachi to make connecting easier. Old pokemon fans used to use nearly the same connection method in VBA-Link, so this method may be well known.

We attempted to connect the same way in the latest master build of VBA-M, but it didn't work. Other people have attempted to get the game to connect with VBA-M in the past, but all attempts have failed.

Even attempting the wireless adapter connection with local mode enabled fails to initiate a match, despite many attempts.

### Expected behavior
First, here's the connection method using VBA-Link which works.
Wireless adapter enabled
Link timeout: 10,000
GBA BIOS not used
Both players share a hamachi server.
One player hosts, the other copies the hamachi ipv4 of the hosting player and uses that to connect.
The hosting player connects as the host ingame, the client connects as client.
![vbal connection](https://user-images.githubusercontent.com/46971717/60388790-1968fd80-9a6b-11e9-86c8-8700b6cd5248.png)

A successful connection then initiates the match.
![connection success](https://user-images.githubusercontent.com/46971717/60388788-1241ef80-9a6b-11e9-97d2-8dd0a72b295e.png)

The host pauses the emulator for a length of time relative to the quality of the connection.
After the match ends, the wireless adapter connection terminates. The emulator must be closed and reopened before making another connection.

### This is the process when attempted in VBA-M
Link type : Wireless
Link timeout: 100,000
no other settings in Options > Link are ticked
GBA BIOS not used

One player hosts, connection for 2, the other copies the hamachi ipv4 of the hosting player and uses that to connect.
The connection menu for the hosting player closes and they receive a join confirmation followed by a disconnect notice.
The client receives this error and fails to connect.
![Capture](https://user-images.githubusercontent.com/46971717/60388797-2c7bcd80-9a6b-11e9-8cac-c0676c9f196e.png)

My partner and I made sure to download a fresh copy of a matching build. We tried connecting many times while switching off between host and client. We also tried toggling "Link at boot" and "Speed hack" and attempting the connection with every possible combination of those toggles. Attempting a Link Cable connection also failed in the same way.

### Summary
Ingame, the connection for wireless adapter mode cannot seem to be made, even when the connection is local.

Attempting to connect over the internet also fails, so there are 2 unique pitfalls for attempting to play BN6 online.

### ROM details
This is specifically for Megaman Battle Network 6: Cybeast Gregar & Falzar.
However, the failure to connect over the internet is not a problem related to any game.

Include screenshots of ROM Information:
![rom information](https://user-images.githubusercontent.com/46971717/60388781-fb030200-9a6a-11e9-99c3-fc6c310a8bfb.png)

### Option details
Video driver: opengl

Sound driver: SDL
### Build details

Operating System:
Windows 10 64bit

Version of code:
2.1.3 64bit master

**Comment by denisfa** [comment]
@ssbmars Thanks for the report. The link code has lots of known issues. We will  see what can be done.

**Comment by retro-wertz** [comment]
making progress, but realized there are at least 2 MMBN6. i had the Gregar here on test, Falzar should probably connect the same.

![Screenshot_2019-07-25_21-39-49](https://user-images.githubusercontent.com/23315382/61882065-4b666600-af2a-11e9-86f6-e8b77e74a459.png)

![Screenshot_2019-07-25_21-40-15](https://user-images.githubusercontent.com/23315382/61881992-240f9900-af2a-11e9-8761-350b10a183ac.png)

**Comment by ssbmars** [comment]
Yeah there are two separate versions, Gregar and Falzar. They connect between each other with no issue. 
Different localizations are incompatible though. There are 3 separate localizations: JP, EU, and international. It can be tricky because both EU and international are in English but they can't connect between each other

**Comment by retro-wertz** [comment]
ok. ill see if i can at least trade with the two versions
